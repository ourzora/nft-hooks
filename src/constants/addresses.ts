import { Networks } from './networks';

export const ZORA_MEDIA_CONTRACT_BY_NETWORK = {
  [Networks.MAINNET]: '0xabEFBc9fD2F806065b4f3C237d4b59D9A97Bcac7',
  [Networks.ROPSTEN]: '0xabEFBc9fD2F806065b4f3C237d4b59D9A97Bcac7',
  [Networks.RINKEBY]: '0x7C2668BD0D3c050703CEcC956C11Bd520c26f7d4',
  [Networks.MUMBAI]: '0xabEFBc9fD2F806065b4f3C237d4b59D9A97Bcac7',
  [Networks.POLYGON]: '0x6953190AAfD8f8995e8f47e8F014d0dB83E92300',
};

export const ENS_REVERSE_LOOKUP_CONTRACT_BY_NETWORK = {
  [Networks.MAINNET]: '0x3671aE578E63FdF66ad4F3E12CC0c0d71Ac7510C',
  [Networks.RINKEBY]: '0x34c01b31c18303Cbd4d97833bb54F70423C4d523',
}