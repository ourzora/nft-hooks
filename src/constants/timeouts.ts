export type RequestType = 'Zora' | 'Graph' | 'IPFS' | 'OpenSea';
export type TimeoutsLookupType = Record<RequestType, number>;
export const DEFAULT_NETWORK_TIMEOUTS_MS: TimeoutsLookupType = {
  Zora: 2 * 1000,
  Graph: 5 * 1000,
  IPFS: 10 * 1000,
  OpenSea: 8 * 1000,
};
