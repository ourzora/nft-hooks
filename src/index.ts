import { Networks, NetworkIDs } from './constants/networks';
import { useNFT, useNFTType } from './hooks/useNFT';
import { useNFTContent, useNFTContentType } from './hooks/useNFTContent';
import { useENSAddress } from './hooks/useENSAddress';
import { useNFTMetadata, useNFTMetadataType } from './hooks/useNFTMetadata';
import { NFTFetchConfiguration } from './context/NFTFetchContext';
import { MediaFetchAgent } from './fetcher/MediaFetchAgent';
import { useZoraUsername } from './hooks/useZoraUsername';
import { RequestError } from './fetcher/RequestError';
import { CurrencyValue, NFTObject } from './types/NFTInterface';
import * as Strategies from './strategies';
import * as Backends from './backends';
import { useNFTQuery } from './hooks/useNFTQuery';

export {
  // Main useNFT hool
  useNFT,
  // Get content of a file. Used by rendering libraries
  useNFTContent,
  // Get metadata for content. Only useful for strategies that don't handle metadata
  useNFTMetadata,
  // Wrapped by useNFT, can use the underlying hooks here
  useZoraUsername,
  // ENS address helper, deprecated Zora username helper
  useENSAddress,
  // Media hook types
  useNFTContentType,
  // Shared standard NFT query hook
  useNFTQuery,
  // Used only for backwards compat with old strategies not providing metadata
  useNFTMetadataType,
  // Used as a return type for useNFTMetadata*
  useNFTType,
  // New types
  // Shared NFT object types
  NFTObject,
  // Fetching strategies
  Strategies,
  // Backends for fetching data (use for debugging or server transforms)
  Backends,
  CurrencyValue,
  // Configuration
  NFTFetchConfiguration,
  // Fetch Agent underlying helper
  MediaFetchAgent,
  // Constants
  Networks,
  // Constant Types
  NetworkIDs,
  // Error type
  RequestError,
};
