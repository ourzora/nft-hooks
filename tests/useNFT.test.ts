import { renderHook } from '@testing-library/react-hooks';
import { Networks, useNFT, useZNFT } from '../src';
import { ZORA_MEDIA_CONTRACT_BY_NETWORK } from '../src/constants/addresses';
import { useOpenseaNFT } from '../src/hooks/useOpenseaNFT';

jest.mock('../src/hooks/useOpenseaNFT');
jest.mock('../src/hooks/useZNFT');

describe('useNFT', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });
  it('fetches opensea NFT', () => {
    renderHook(() => useNFT('0x00000000120040', '23'));
    expect(useOpenseaNFT).toHaveBeenLastCalledWith('0x00000000120040', '23', {});
    expect(useOpenseaNFT).toHaveBeenCalledTimes(1);
    expect(useZNFT).toHaveBeenLastCalledWith(undefined, {});
    expect(useOpenseaNFT).toHaveBeenCalledTimes(1);
  });
  it('fetches zora NFT', () => {
    renderHook(() => useNFT(undefined, '23'));
    expect(useZNFT).toHaveBeenCalledWith('23', {});
    expect(useZNFT).toHaveBeenCalledTimes(1);
    expect(useOpenseaNFT).toHaveBeenCalledWith(undefined, undefined, {});
    expect(useOpenseaNFT).toHaveBeenCalledTimes(1);
  });
  it('fetches zora NFT by address', () => {
    renderHook(() => useNFT(ZORA_MEDIA_CONTRACT_BY_NETWORK[Networks.MAINNET], '23'));
    expect(useZNFT).toHaveBeenCalledWith("23", {});
    expect(useOpenseaNFT).toHaveBeenCalledWith(undefined, undefined, {});
    expect(useZNFT).toHaveBeenCalledTimes(1);
    expect(useOpenseaNFT).toHaveBeenCalledTimes(1);
  });
});
