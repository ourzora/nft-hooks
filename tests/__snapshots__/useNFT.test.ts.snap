// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useNFT caches multiple NFTs being loaded 1`] = `
Object {
  "auction": Object {
    "current": Object {
      "auctionType": "perpetual",
      "endingAt": undefined,
      "likelyHasEnded": false,
      "reserveMet": false,
      "reservePrice": undefined,
    },
    "highestBid": Object {
      "placedAt": "12974",
      "placedBy": "111",
      "pricing": Object {
        "amount": "12974",
        "computedValue": undefined,
        "currency": Object {
          "decimals": 18,
          "id": "0xFACE",
          "name": "Wrapped Ether",
          "symbol": "WETH",
        },
        "prettyAmount": "1.2974e-14",
      },
    },
  },
  "nft": Object {
    "contentHash": "ByTeSStrInG",
    "contentURI": "Hello World",
    "creator": Object {
      "id": "107",
    },
    "id": "1",
    "metadataHash": "ByTeSStrInG",
    "metadataURI": "Hello World",
    "owner": Object {
      "id": "106",
    },
  },
  "pricing": Object {
    "perpetual": Object {
      "ask": Object {
        "createdAtTimestamp": "12974",
        "id": "108",
        "pricing": Object {
          "amount": "12974",
          "computedValue": undefined,
          "currency": Object {
            "decimals": 18,
            "id": "0xFACE",
            "name": "Wrapped Ether",
            "symbol": "WETH",
          },
          "prettyAmount": "1.2974e-14",
        },
      },
      "bids": Array [
        Object {
          "bidder": Object {
            "id": "111",
          },
          "createdAtTimestamp": "12974",
          "id": "109",
          "pricing": Object {
            "amount": "12974",
            "computedValue": undefined,
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
        Object {
          "bidder": Object {
            "id": "112",
          },
          "createdAtTimestamp": "12974",
          "id": "110",
          "pricing": Object {
            "amount": "12974",
            "computedValue": undefined,
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
      ],
    },
    "reserve": undefined,
  },
}
`;

exports[`useNFT caches multiple NFTs being loaded 2`] = `
Object {
  "auction": Object {
    "current": Object {
      "auctionType": "perpetual",
      "endingAt": undefined,
      "likelyHasEnded": false,
      "reserveMet": false,
      "reservePrice": undefined,
    },
    "highestBid": Object {
      "placedAt": "12974",
      "placedBy": "118",
      "pricing": Object {
        "amount": "12974",
        "computedValue": undefined,
        "currency": Object {
          "decimals": 18,
          "id": "0xFACE",
          "name": "Wrapped Ether",
          "symbol": "WETH",
        },
        "prettyAmount": "1.2974e-14",
      },
    },
  },
  "nft": Object {
    "contentHash": "ByTeSStrInG",
    "contentURI": "Hello World",
    "creator": Object {
      "id": "114",
    },
    "id": "2",
    "metadataHash": "ByTeSStrInG",
    "metadataURI": "Hello World",
    "owner": Object {
      "id": "113",
    },
  },
  "pricing": Object {
    "perpetual": Object {
      "ask": Object {
        "createdAtTimestamp": "12974",
        "id": "115",
        "pricing": Object {
          "amount": "12974",
          "computedValue": undefined,
          "currency": Object {
            "decimals": 18,
            "id": "0xFACE",
            "name": "Wrapped Ether",
            "symbol": "WETH",
          },
          "prettyAmount": "1.2974e-14",
        },
      },
      "bids": Array [
        Object {
          "bidder": Object {
            "id": "118",
          },
          "createdAtTimestamp": "12974",
          "id": "116",
          "pricing": Object {
            "amount": "12974",
            "computedValue": undefined,
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
        Object {
          "bidder": Object {
            "id": "119",
          },
          "createdAtTimestamp": "12974",
          "id": "117",
          "pricing": Object {
            "amount": "12974",
            "computedValue": undefined,
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
      ],
    },
    "reserve": undefined,
  },
}
`;

exports[`useNFT caches multiple NFTs being loaded 3`] = `
Object {
  "auction": Object {
    "current": Object {
      "auctionType": "perpetual",
      "endingAt": undefined,
      "likelyHasEnded": false,
      "reserveMet": false,
      "reservePrice": undefined,
    },
    "highestBid": Object {
      "placedAt": "12974",
      "placedBy": "111",
      "pricing": Object {
        "amount": "12974",
        "computedValue": undefined,
        "currency": Object {
          "decimals": 18,
          "id": "0xFACE",
          "name": "Wrapped Ether",
          "symbol": "WETH",
        },
        "prettyAmount": "1.2974e-14",
      },
    },
  },
  "nft": Object {
    "contentHash": "ByTeSStrInG",
    "contentURI": "Hello World",
    "creator": Object {
      "id": "107",
    },
    "id": "1",
    "metadataHash": "ByTeSStrInG",
    "metadataURI": "Hello World",
    "owner": Object {
      "id": "106",
    },
  },
  "pricing": Object {
    "perpetual": Object {
      "ask": Object {
        "createdAtTimestamp": "12974",
        "id": "108",
        "pricing": Object {
          "amount": "12974",
          "computedValue": undefined,
          "currency": Object {
            "decimals": 18,
            "id": "0xFACE",
            "name": "Wrapped Ether",
            "symbol": "WETH",
          },
          "prettyAmount": "1.2974e-14",
        },
      },
      "bids": Array [
        Object {
          "bidder": Object {
            "id": "111",
          },
          "createdAtTimestamp": "12974",
          "id": "109",
          "pricing": Object {
            "amount": "12974",
            "computedValue": undefined,
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
        Object {
          "bidder": Object {
            "id": "112",
          },
          "createdAtTimestamp": "12974",
          "id": "110",
          "pricing": Object {
            "amount": "12974",
            "computedValue": undefined,
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
      ],
    },
    "reserve": undefined,
  },
}
`;

exports[`useNFT caches multiple NFTs being loaded 4`] = `
Object {
  "auction": Object {
    "current": Object {
      "auctionType": "perpetual",
      "endingAt": undefined,
      "likelyHasEnded": false,
      "reserveMet": false,
      "reservePrice": undefined,
    },
    "highestBid": Object {
      "placedAt": "12974",
      "placedBy": "118",
      "pricing": Object {
        "amount": "12974",
        "computedValue": undefined,
        "currency": Object {
          "decimals": 18,
          "id": "0xFACE",
          "name": "Wrapped Ether",
          "symbol": "WETH",
        },
        "prettyAmount": "1.2974e-14",
      },
    },
  },
  "nft": Object {
    "contentHash": "ByTeSStrInG",
    "contentURI": "Hello World",
    "creator": Object {
      "id": "114",
    },
    "id": "2",
    "metadataHash": "ByTeSStrInG",
    "metadataURI": "Hello World",
    "owner": Object {
      "id": "113",
    },
  },
  "pricing": Object {
    "perpetual": Object {
      "ask": Object {
        "createdAtTimestamp": "12974",
        "id": "115",
        "pricing": Object {
          "amount": "12974",
          "computedValue": undefined,
          "currency": Object {
            "decimals": 18,
            "id": "0xFACE",
            "name": "Wrapped Ether",
            "symbol": "WETH",
          },
          "prettyAmount": "1.2974e-14",
        },
      },
      "bids": Array [
        Object {
          "bidder": Object {
            "id": "118",
          },
          "createdAtTimestamp": "12974",
          "id": "116",
          "pricing": Object {
            "amount": "12974",
            "computedValue": undefined,
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
        Object {
          "bidder": Object {
            "id": "119",
          },
          "createdAtTimestamp": "12974",
          "id": "117",
          "pricing": Object {
            "amount": "12974",
            "computedValue": undefined,
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
      ],
    },
    "reserve": undefined,
  },
}
`;

exports[`useNFT correctly loads auction information from uniswap 1`] = `
Object {
  "auction": Object {
    "current": Object {
      "auctionType": "reserve",
      "endingAt": "12974",
      "likelyHasEnded": false,
      "reserveMet": false,
      "reservePrice": Object {
        "amount": "12974",
        "computedValue": Object {
          "inETH": "180004289.8602",
          "inUSD": "2497421332516.94932446",
        },
        "currency": Object {
          "decimals": 18,
          "id": "0xFACE",
          "name": "Wrapped Ether",
          "symbol": "WETH",
        },
        "prettyAmount": "1.2974e-14",
      },
    },
    "highestBid": Object {
      "placedAt": "12974",
      "placedBy": "29",
      "pricing": Object {
        "amount": "12974",
        "computedValue": Object {
          "inETH": "180004289.8602",
          "inUSD": "2497421332516.94932446",
        },
        "currency": Object {
          "decimals": 18,
          "id": "0xFACE",
          "name": "Wrapped Ether",
          "symbol": "WETH",
        },
        "prettyAmount": "1.2974e-14",
      },
    },
  },
  "nft": Object {
    "contentHash": "ByTeSStrInG",
    "contentURI": "https://zora.co/content",
    "creator": Object {
      "id": "23",
    },
    "id": "2974",
    "metadataHash": "ByTeSStrInG",
    "metadataURI": "https://zora.co/content",
    "owner": Object {
      "id": "22",
    },
  },
  "pricing": Object {
    "perpetual": Object {
      "ask": Object {
        "createdAtTimestamp": "12974",
        "id": "24",
        "pricing": Object {
          "amount": "12974",
          "computedValue": Object {
            "inETH": "180004289.8602",
            "inUSD": "2497421332516.94932446",
          },
          "currency": Object {
            "decimals": 18,
            "id": "0xFACE",
            "name": "Wrapped Ether",
            "symbol": "WETH",
          },
          "prettyAmount": "1.2974e-14",
        },
      },
      "bids": Array [
        Object {
          "bidder": Object {
            "id": "10",
          },
          "createdAtTimestamp": "12974",
          "id": "25",
          "pricing": Object {
            "amount": "10000",
            "computedValue": Object {
              "inETH": "138742323",
              "inUSD": "1924943219143.6329",
            },
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1e-14",
          },
        },
      ],
    },
    "reserve": Object {
      "auctionCurrency": Object {
        "decimals": 18,
        "id": "0xFACE",
        "name": "Wrapped Ether",
        "symbol": "WETH",
      },
      "createdAtTimestamp": "12974",
      "currentBid": Object {
        "amount": "12974",
        "bidType": "Final",
        "bidder": Object {
          "id": "29",
        },
        "createdAtTimestamp": "12974",
      },
      "duration": "12974",
      "expectedEndTimestamp": "12974",
      "finalizedAtTimestamp": "12974",
      "firstBidTime": "12974",
      "id": "26",
      "previousBids": Array [
        Object {
          "bidInactivatedAtBlockNumber": "12974",
          "bidInactivatedAtTimestamp": "12974",
          "bidType": "Final",
          "bidder": Object {
            "id": "32",
          },
          "createdAtTimestamp": "12974",
          "id": "30",
          "pricing": Object {
            "amount": "12974",
            "computedValue": Object {
              "inETH": "180004289.8602",
              "inUSD": "2497421332516.94932446",
            },
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
        Object {
          "bidInactivatedAtBlockNumber": "12974",
          "bidInactivatedAtTimestamp": "12974",
          "bidType": "Final",
          "bidder": Object {
            "id": "33",
          },
          "createdAtTimestamp": "12974",
          "id": "31",
          "pricing": Object {
            "amount": "12974",
            "computedValue": Object {
              "inETH": "180004289.8602",
              "inUSD": "2497421332516.94932446",
            },
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
      ],
      "reservePrice": "1000000",
      "status": "Active",
      "tokenId": "2974",
    },
  },
}
`;

exports[`useNFT correctly loads auction information from uniswap 2`] = `
Object {
  "auction": Object {
    "current": Object {
      "auctionType": "reserve",
      "endingAt": "12974",
      "likelyHasEnded": false,
      "reserveMet": false,
      "reservePrice": Object {
        "amount": "12974",
        "computedValue": Object {
          "inETH": "180004289.8602",
          "inUSD": "2497421332516.94932446",
        },
        "currency": Object {
          "decimals": 18,
          "id": "0xFACE",
          "name": "Wrapped Ether",
          "symbol": "WETH",
        },
        "prettyAmount": "1.2974e-14",
      },
    },
    "highestBid": Object {
      "placedAt": "12974",
      "placedBy": "29",
      "pricing": Object {
        "amount": "12974",
        "computedValue": Object {
          "inETH": "180004289.8602",
          "inUSD": "2497421332516.94932446",
        },
        "currency": Object {
          "decimals": 18,
          "id": "0xFACE",
          "name": "Wrapped Ether",
          "symbol": "WETH",
        },
        "prettyAmount": "1.2974e-14",
      },
    },
  },
  "nft": Object {
    "contentHash": "ByTeSStrInG",
    "contentURI": "https://zora.co/content",
    "creator": Object {
      "id": "23",
    },
    "id": "2974",
    "metadataHash": "ByTeSStrInG",
    "metadataURI": "https://zora.co/content",
    "owner": Object {
      "id": "22",
    },
  },
  "pricing": Object {
    "perpetual": Object {
      "ask": Object {
        "createdAtTimestamp": "12974",
        "id": "24",
        "pricing": Object {
          "amount": "12974",
          "computedValue": Object {
            "inETH": "180004289.8602",
            "inUSD": "2497421332516.94932446",
          },
          "currency": Object {
            "decimals": 18,
            "id": "0xFACE",
            "name": "Wrapped Ether",
            "symbol": "WETH",
          },
          "prettyAmount": "1.2974e-14",
        },
      },
      "bids": Array [
        Object {
          "bidder": Object {
            "id": "10",
          },
          "createdAtTimestamp": "12974",
          "id": "25",
          "pricing": Object {
            "amount": "10000",
            "computedValue": Object {
              "inETH": "138742323",
              "inUSD": "1924943219143.6329",
            },
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1e-14",
          },
        },
      ],
    },
    "reserve": Object {
      "auctionCurrency": Object {
        "decimals": 18,
        "id": "0xFACE",
        "name": "Wrapped Ether",
        "symbol": "WETH",
      },
      "createdAtTimestamp": "12974",
      "currentBid": Object {
        "amount": "12974",
        "bidType": "Final",
        "bidder": Object {
          "id": "29",
        },
        "createdAtTimestamp": "12974",
      },
      "duration": "12974",
      "expectedEndTimestamp": "12974",
      "finalizedAtTimestamp": "12974",
      "firstBidTime": "12974",
      "id": "26",
      "previousBids": Array [
        Object {
          "bidInactivatedAtBlockNumber": "12974",
          "bidInactivatedAtTimestamp": "12974",
          "bidType": "Final",
          "bidder": Object {
            "id": "32",
          },
          "createdAtTimestamp": "12974",
          "id": "30",
          "pricing": Object {
            "amount": "12974",
            "computedValue": Object {
              "inETH": "180004289.8602",
              "inUSD": "2497421332516.94932446",
            },
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
        Object {
          "bidInactivatedAtBlockNumber": "12974",
          "bidInactivatedAtTimestamp": "12974",
          "bidType": "Final",
          "bidder": Object {
            "id": "33",
          },
          "createdAtTimestamp": "12974",
          "id": "31",
          "pricing": Object {
            "amount": "12974",
            "computedValue": Object {
              "inETH": "180004289.8602",
              "inUSD": "2497421332516.94932446",
            },
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
      ],
      "reservePrice": "1000000",
      "status": "Active",
      "tokenId": "2974",
    },
  },
}
`;

exports[`useNFT correctly loads multiple perpetual bid NFTs 1`] = `
Object {
  "auction": Object {
    "current": Object {
      "auctionType": "perpetual",
      "endingAt": undefined,
      "likelyHasEnded": false,
      "reserveMet": false,
      "reservePrice": undefined,
    },
    "highestBid": Object {
      "placedAt": "12974",
      "placedBy": "83",
      "pricing": Object {
        "amount": "12974",
        "computedValue": undefined,
        "currency": Object {
          "decimals": 18,
          "id": "0xFACE",
          "name": "Wrapped Ether",
          "symbol": "WETH",
        },
        "prettyAmount": "1.2974e-14",
      },
    },
  },
  "nft": Object {
    "contentHash": "ByTeSStrInG",
    "contentURI": "Hello World",
    "creator": Object {
      "id": "79",
    },
    "id": "1",
    "metadataHash": "ByTeSStrInG",
    "metadataURI": "Hello World",
    "owner": Object {
      "id": "78",
    },
  },
  "pricing": Object {
    "perpetual": Object {
      "ask": Object {
        "createdAtTimestamp": "12974",
        "id": "80",
        "pricing": Object {
          "amount": "12974",
          "computedValue": undefined,
          "currency": Object {
            "decimals": 18,
            "id": "0xFACE",
            "name": "Wrapped Ether",
            "symbol": "WETH",
          },
          "prettyAmount": "1.2974e-14",
        },
      },
      "bids": Array [
        Object {
          "bidder": Object {
            "id": "83",
          },
          "createdAtTimestamp": "12974",
          "id": "81",
          "pricing": Object {
            "amount": "12974",
            "computedValue": undefined,
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
        Object {
          "bidder": Object {
            "id": "84",
          },
          "createdAtTimestamp": "12974",
          "id": "82",
          "pricing": Object {
            "amount": "12974",
            "computedValue": undefined,
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
      ],
    },
    "reserve": undefined,
  },
}
`;

exports[`useNFT correctly loads multiple perpetual bid NFTs 2`] = `
Object {
  "auction": Object {
    "current": Object {
      "auctionType": "perpetual",
      "endingAt": undefined,
      "likelyHasEnded": false,
      "reserveMet": false,
      "reservePrice": undefined,
    },
    "highestBid": Object {
      "placedAt": "12974",
      "placedBy": "90",
      "pricing": Object {
        "amount": "12974",
        "computedValue": undefined,
        "currency": Object {
          "decimals": 18,
          "id": "0xFACE",
          "name": "Wrapped Ether",
          "symbol": "WETH",
        },
        "prettyAmount": "1.2974e-14",
      },
    },
  },
  "nft": Object {
    "contentHash": "ByTeSStrInG",
    "contentURI": "Hello World",
    "creator": Object {
      "id": "86",
    },
    "id": "2",
    "metadataHash": "ByTeSStrInG",
    "metadataURI": "Hello World",
    "owner": Object {
      "id": "85",
    },
  },
  "pricing": Object {
    "perpetual": Object {
      "ask": Object {
        "createdAtTimestamp": "12974",
        "id": "87",
        "pricing": Object {
          "amount": "12974",
          "computedValue": undefined,
          "currency": Object {
            "decimals": 18,
            "id": "0xFACE",
            "name": "Wrapped Ether",
            "symbol": "WETH",
          },
          "prettyAmount": "1.2974e-14",
        },
      },
      "bids": Array [
        Object {
          "bidder": Object {
            "id": "90",
          },
          "createdAtTimestamp": "12974",
          "id": "88",
          "pricing": Object {
            "amount": "12974",
            "computedValue": undefined,
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
        Object {
          "bidder": Object {
            "id": "91",
          },
          "createdAtTimestamp": "12974",
          "id": "89",
          "pricing": Object {
            "amount": "12974",
            "computedValue": undefined,
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
      ],
    },
    "reserve": undefined,
  },
}
`;

exports[`useNFT loads an NFT not in an auction with bids 1`] = `
Object {
  "auction": Object {
    "current": Object {
      "auctionType": "perpetual",
      "endingAt": undefined,
      "likelyHasEnded": false,
      "reserveMet": false,
      "reservePrice": undefined,
    },
    "highestBid": Object {
      "placedAt": "12974",
      "placedBy": "10",
      "pricing": Object {
        "amount": "10000",
        "computedValue": undefined,
        "currency": Object {
          "decimals": 18,
          "id": "0xFACE",
          "name": "Wrapped Ether",
          "symbol": "WETH",
        },
        "prettyAmount": "1e-14",
      },
    },
  },
  "nft": Object {
    "contentHash": "ByTeSStrInG",
    "contentURI": "https://zora.co/content",
    "creator": Object {
      "id": "44",
    },
    "id": "2974",
    "metadataHash": "ByTeSStrInG",
    "metadataURI": "https://zora.co/content",
    "owner": Object {
      "id": "43",
    },
  },
  "pricing": Object {
    "perpetual": Object {
      "ask": Object {
        "createdAtTimestamp": "12974",
        "id": "45",
        "pricing": Object {
          "amount": "12974",
          "computedValue": undefined,
          "currency": Object {
            "decimals": 18,
            "id": "0xFACE",
            "name": "Wrapped Ether",
            "symbol": "WETH",
          },
          "prettyAmount": "1.2974e-14",
        },
      },
      "bids": Array [
        Object {
          "bidder": Object {
            "id": "10",
          },
          "createdAtTimestamp": "12974",
          "id": "46",
          "pricing": Object {
            "amount": "10000",
            "computedValue": undefined,
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1e-14",
          },
        },
      ],
    },
    "reserve": undefined,
  },
}
`;

exports[`useNFT loads an NFT with no bids and no auction 1`] = `
Object {
  "auction": Object {
    "current": Object {
      "auctionType": "perpetual",
      "endingAt": undefined,
      "likelyHasEnded": false,
      "reserveMet": false,
      "reservePrice": undefined,
    },
    "highestBid": undefined,
  },
  "nft": Object {
    "contentHash": "ByTeSStrInG",
    "contentURI": "https://zora.co/content",
    "creator": Object {
      "id": "62",
    },
    "id": "2974",
    "metadataHash": "ByTeSStrInG",
    "metadataURI": "https://zora.co/content",
    "owner": Object {
      "id": "61",
    },
  },
  "pricing": Object {
    "perpetual": Object {
      "ask": Object {
        "createdAtTimestamp": "12974",
        "id": "63",
        "pricing": Object {
          "amount": "12974",
          "computedValue": undefined,
          "currency": Object {
            "decimals": 18,
            "id": "0xFACE",
            "name": "Wrapped Ether",
            "symbol": "WETH",
          },
          "prettyAmount": "1.2974e-14",
        },
      },
      "bids": Array [],
    },
    "reserve": undefined,
  },
}
`;

exports[`useNFT loads an nft currently in an auction 1`] = `
Object {
  "auction": Object {
    "current": Object {
      "auctionType": "reserve",
      "endingAt": "12974",
      "likelyHasEnded": false,
      "reserveMet": false,
      "reservePrice": Object {
        "amount": "12974",
        "computedValue": undefined,
        "currency": Object {
          "decimals": 18,
          "id": "0xFACE",
          "name": "Wrapped Ether",
          "symbol": "WETH",
        },
        "prettyAmount": "1.2974e-14",
      },
    },
    "highestBid": Object {
      "placedAt": "12974",
      "placedBy": "9",
      "pricing": Object {
        "amount": "12974",
        "computedValue": undefined,
        "currency": Object {
          "decimals": 18,
          "id": "0xFACE",
          "name": "Wrapped Ether",
          "symbol": "WETH",
        },
        "prettyAmount": "1.2974e-14",
      },
    },
  },
  "nft": Object {
    "contentHash": "ByTeSStrInG",
    "contentURI": "https://zora.co/content",
    "creator": Object {
      "id": "3",
    },
    "id": "2974",
    "metadataHash": "ByTeSStrInG",
    "metadataURI": "https://zora.co/content",
    "owner": Object {
      "id": "2",
    },
  },
  "pricing": Object {
    "perpetual": Object {
      "ask": Object {
        "createdAtTimestamp": "12974",
        "id": "4",
        "pricing": Object {
          "amount": "12974",
          "computedValue": undefined,
          "currency": Object {
            "decimals": 18,
            "id": "0xFACE",
            "name": "Wrapped Ether",
            "symbol": "WETH",
          },
          "prettyAmount": "1.2974e-14",
        },
      },
      "bids": Array [
        Object {
          "bidder": Object {
            "id": "10",
          },
          "createdAtTimestamp": "12974",
          "id": "5",
          "pricing": Object {
            "amount": "10000",
            "computedValue": undefined,
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1e-14",
          },
        },
      ],
    },
    "reserve": Object {
      "auctionCurrency": Object {
        "decimals": 18,
        "id": "0xFACE",
        "name": "Wrapped Ether",
        "symbol": "WETH",
      },
      "createdAtTimestamp": "12974",
      "currentBid": Object {
        "amount": "12974",
        "bidType": "Final",
        "bidder": Object {
          "id": "9",
        },
        "createdAtTimestamp": "12974",
      },
      "duration": "12974",
      "expectedEndTimestamp": "12974",
      "finalizedAtTimestamp": "12974",
      "firstBidTime": "12974",
      "id": "6",
      "previousBids": Array [
        Object {
          "bidInactivatedAtBlockNumber": "12974",
          "bidInactivatedAtTimestamp": "12974",
          "bidType": "Final",
          "bidder": Object {
            "id": "12",
          },
          "createdAtTimestamp": "12974",
          "id": "10",
          "pricing": Object {
            "amount": "12974",
            "computedValue": undefined,
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
        Object {
          "bidInactivatedAtBlockNumber": "12974",
          "bidInactivatedAtTimestamp": "12974",
          "bidType": "Final",
          "bidder": Object {
            "id": "13",
          },
          "createdAtTimestamp": "12974",
          "id": "11",
          "pricing": Object {
            "amount": "12974",
            "computedValue": undefined,
            "currency": Object {
              "decimals": 18,
              "id": "0xFACE",
              "name": "Wrapped Ether",
              "symbol": "WETH",
            },
            "prettyAmount": "1.2974e-14",
          },
        },
      ],
      "reservePrice": "1000000",
      "status": "Active",
      "tokenId": "2974",
    },
  },
}
`;
